core/src/router/base.ts

```
@@ -401,7 +402,7 @@ export abstract class BaseActionRouter<K extends ActionKind> extends BaseRouter
     )
     const spec = this.definitions[actionType]
 
-    if (handlers.length === 0 && spec.base && !pluginName) {
+    if (handlers.length === 0 && spec?.base && !pluginName) {
```

Alternatively, we can check if the spec is defined and throw an error:
```
if (!spec) {
      throw new ConfigurationError(
        `Cannot find handler "${String(
          handlerType
        )}" for action type "${actionType}". Available handlers: [${Object.keys(this.handlers[handlerType]).join(
          ","
        )}]`,
        {
          handlerType,
          actionType,
        }
      )
    }
```

But I'm not sure if that would be expected behavior.

---

I think it makes sense to throw here if the spec isn't found

---

Alright, thanks! I'll change this. Should we throw an error in the rest places as well?

---

Ah, the error is thrown later in [L482](https://github.com/garden-io/garden/blob/d63595c3ad9f2ff1065907d9b281b9ac892d4fca/core/src/router/base.ts#L482) is there is no default handler provided. So, maybe it's ok to leave it as it is now, wdyt @edvald ?